lib/handler.php

```diff
@@ -48,4 +48,112 @@ public static function addBadge(rex_extension_point $ep)
+
+                continue;
+            }
+
+            if (!is_array($subpages)) {
+                self::protectPage($page);
+
+                continue;
+            }
+
+            foreach ($subpages as $subpage) {
+                $subpage = $page->getSubpage($subpage);
+
+                if ($subpage) {
+                    self::protectPage($subpage);
+                }
+            }
+        }
+    }
+
+    public static function getProtectedPages(): array
+    {
+        return rex_addon::get('ydeploy')->getProperty('config')['protected_pages'];
+    }
+
+    public static function getUnlockedPages(): array
+    {
+        return rex_session('ydeploy_unlocked_pages', 'array', []);
+    }
+
+    public static function unlockPage(string $page)
+    {
+        $unlockedPages = self::getUnlockedPages();
+        $unlockedPages[$page] = true;
+        rex_set_session('ydeploy_unlocked_pages', $unlockedPages);
+    }
+
+    public static function lockPage(string $page)
+    {
+        $unlockedPages = self::getUnlockedPages();
+        unset($unlockedPages[$page]);
+        rex_set_session('ydeploy_unlocked_pages', $unlockedPages);
+    }
+
+    private static function protectPage(rex_be_page $page)
+    {
+        if (rex_be_controller::getCurrentPage() && $page->isActive()) {
+            rex_be_controller::setCurrentPage('system/ydeploy');
+        }
+
+        $page->setHidden(true);
+
+        // If page is first subpage of other page, then the other page must be also hidden
+        while ($parent = $page->getParent()) {
+            $subpages = $parent->getSubpages();
+
+            if ($page !== reset($subpages)) {
+                break;
+            }
+
+            $parent->setHidden(true);
+            $page = $parent;
+        }
+    }
+
+    private static function handleUnlockedPage(rex_be_page $page, array $subpages = null)
+    {
+        if (!rex_be_controller::getCurrentPage() || !$page->isActive()) {
+            return;
+        }
+
+        if (is_array($subpages)) {
+            $subpage = substr(rex_be_controller::getCurrentPage(), strlen($page->getFullKey()) + 1);
+            $subpage = explode('/', $subpage, 2)[0];
+
+            if (!in_array($subpage, $subpages, true)) {
+                return;
+            }
+        }
+
+        rex_extension::register('PAGE_TITLE_SHOWN', function (rex_extension_point $ep) {
+            $url = rex_url::backendPage('system/ydeploy', rex_api_ydeploy_protected_page::getUrlParams() + [
+                'action' => 'lock',
+                'protected_page' => rex_be_controller::getCurrentPage(),
+            ]);
+            $error = rex_view::error('
+                    The page <code>'.rex_escape(rex_be_controller::getCurrentPage()).'</code> is protected in deployed instances, but currently unlocked. Changes via this page should be made in development instances only! <br><br>
+                    
+                    <a href="'.$url.'">Lock and leave this page</a>
```

Comment:
hier nochmal "this page"

Reply:
Wobei ich das an der Stelle eigentlich ok finde, direkt darüber wird ja die Page genannt.

Reply:
So Sieht es zurzeit aus:

![screenshot 2018-03-20 12 12 00](https://user-images.githubusercontent.com/330436/37651293-f33fac00-2c37-11e8-8e53-b5ff10681a36.png)